pkg/webhook/admission/http.go

```diff
@@ -124,8 +124,16 @@ func (wh *Webhook) writeResponseTyped(w io.Writer, response Response, admRevGVK
 // writeAdmissionResponse writes ar to w.
 func (wh *Webhook) writeAdmissionResponse(w io.Writer, ar v1.AdmissionReview) {
 	if err := json.NewEncoder(w).Encode(ar); err != nil {
-		wh.log.Error(err, "unable to encode the response")
-		wh.writeResponse(w, Errored(http.StatusInternalServerError, err))
+		wh.log.Error(err, "unable to encode and write the response")
+		// Since the `ar v1.AdmissionReview` is a clear and legal object,
+		// it should not have problem to be marshalled into bytes.
+		// The error here is probably caused by the abnormal HTTP connection,
+		// e.g., broken pipe, so we can only write the error response once,
+		// to avoid endless circular calling.
```

Comment:
If the connection has broken, it will run into a dead loop:

1. `json.NewEncoder(w).Encode(ar)` failed because of `broken pipe`
2.  call `wh.writeResponse(w, Errored(http.StatusInternalServerError, err))`
3. call `wh.writeAdmissionResponse(w, v1.AdmissionReview{Response: &response.AdmissionResponse})`
4. again `json.NewEncoder(w).Encode(ar)` failed because of `broken pipe`
5. ...

Reply:
@alvaroaleman WDYT :)

Reply:
Ah, thanks for the explanation. Looks good to me, but can we add a test with a broken `io.Writer` to test this scenario we are fixing?

Reply:
Test added.